README


This archive contains the distribution Email-Sender,
version 1.300016:

  a library for sending email

This software is copyright (c) 2014 by Ricardo Signes.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.020.



Changes
Revision history for Email-Sender

1.300016  2014-10-10 22:54:00-04:00 America/New_York
        - no changes since 1.300015, just made non-trial release

1.300015  2014-09-12 22:01:37-04:00 America/New_York (TRIAL RELEASE)
          [THIS MIGHT BREAK YOUR CODE]
        - version 1.300008 was meant to rewrite line endings when sending to
          sendmail, but the change was not effective; it has been fixed

1.300014  2014-08-28 14:21:26-04:00 America/New_York
        - do not try passing an empty string as a Failure message

1.300013  2014-08-28 11:00:16-04:00 America/New_York
        - sometimes, if no error is given and $smtp->message returns (),
          the arguments to the Failure constructor were out of alignment and
          the failure's error message would be "code"; this has been fixed

1.300012  2014-07-10 09:55:35-04:00 America/New_York
        - documentation fix: there's no Email::MIME->delete_header
          (thanks, Kris Matthews)

1.300011  2014-03-02 18:16:12-05:00 America/New_York
        - run the sendmail program with -i by default (thanks, William Blunn)
        - do not die when a race causes mkpath(Maildir/new) to fail because the
          directory already exists

1.300010  2014-01-11 14:15:13-05:00 America/New_York
        - the SMTP transport now has a "debug" option

1.300009  2013-09-01 12:03:06 America/New_York
          see also 1.300008!

          switch to automatic determination of prereqs to improve accuracy

1.300008  2013-09-01 09:39:13 America/New_York
          [THIS MIGHT BREAK YOUR CODE]
          at long last, Sendmail and Maildir transports replaces CRLF with CR
          before piping

          [BUG FIXES]
          fixes a bug where recipients beginning with a - would not work with
          the Sendmail transport [rt.perl.org #66246]

          require Email-Abstract 3.006, to get Email-Simple 1.998, to get
          Email::Simple->new(\$str), resolving [rt.perl.org #85926]

1.300007  2013-03-19 14:58:27 America/New_York
          when sending over SMTP, send DATA in hunks of 1 mebibyte or smaller

1.300006  2013-03-17 17:23:21 America/New_York
          re-fix the problem with attribute accessors and role methods
          clashing, previously fixed in 0.102360

1.300005  2013-02-14 11:37:09 America/New_York
          add missing prereq, Sub::Exporter; thanks for report, Esteban
          Manchado Velázquez

1.300004  2013-02-09 10:09:02 America/New_York
          bump required version of MooX::Types::MooseLike

1.300003  2013-02-06 15:56:53 America/New_York
          make Wrapper transports proxy is_simple and allow_partial_success to
          the wrapped transport

          allow Wrapper transports to construct their own targets; useful for
          configuring entire wrapper chain from the environment

1.300002  2013-02-06 14:06:24 America/New_York
          THIS IS A BIG DEAL:  see v1.300000

          this is the first non-trial release using Moo instead of Moose

          increase version of Moo to interoperate with ancient Mouse.pm

1.300001  2013-01-07 21:38:31 America/New_York
          increase required versions of Throwable and Moo

1.300000  2013-01-02 22:42:58 America/New_York
          THIS IS A BIG DEAL:

          use Moo instead of Moose (thanks to Justin Hunter and Christian
          Walde!)

0.120002  2012-09-11 14:43:07 America/New_York
          added shift_deliveries to the Test transport, for treating the
          sent messages like a queue of things to inspect

0.120001  2012-07-19 16:28:01 America/Los_Angeles
          no changes from 0.120000, just marked as stable release

0.120000  2012-05-08 20:53:43 America/New_York
          THIS IS A BIG DEAL:

          This release deprecates the scalar context behavior of:

          * Failure->recipients
          * Failure::Multi->failures
          * Failure::Multi->recipients
          * Transport::Test->deliveries
          * Transport::Failable->failure_conditions

          Until now, these methods returned a list in list context and
          an array reference in scalar context.  This still works, but
          scalar context calls will now generate a warning.  In one
          year (2013-05), they will be become fatal.

0.110005  2012-03-05 21:37:35 America/New_York
          binmode output handle to avoid line ending munging in Mbox and
          Maildir; thanks, Christian Walde!

0.110004  2012-02-21 17:47:35 America/New_York
          import the "Bcc" page from the (otherwise empty) github wiki; this
          adds some information to the QuickStart page about why Bcc support
          isn't there and why this isn't a bug

0.110003  2012-02-01 15:09:52 America/New_York
          add the add_lines_header and add_envelope_headers attributes to
          Maildir transports

0.110002  2012-01-31 17:02:35 America/New_York
          Maildir deliveries now have a ->filename method to tell you just
          where the mail was delivered

0.110001  2011-04-03 16:32:15 America/New_York

          prevent Test::MinimumVersion test from causing a requirement on T::MV
          and, thus, PPI!  Thanks for the report, Matt S. Trout!

0.110000  2011-03-17 21:40:36 America/New_York
          provide a much clearer and more fatal error when SASL authentication
          has failed because Authen::SASL or MIME::Base64 is not available

0.102370  2010-08-25 08:37:22 America/New_York
          remove spurious prereq on Sys::Hostname::Long (thanks ABH)
          tweak some tests for truth/definedness in SMTP transport

0.102360  2010-08-24 07:47:24 America/New_York
          avoid the need to use "excludes" in role application; this silences
          warnings with Moose 1.10

0.101760  2010-06-25 08:18:26 America/New_York
          fix sendmail.t with latest Capture::Tiny (Justin Hunter)

0.100460  2010-02-15 12:58:18 America/New_York
          avoid using AutoPrereq, as it brings in build_requires as requires

0.100450  2010-02-14 17:55:48 America/New_York

Makefile.PL

# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.020.
use strict;
use warnings;



use ExtUtils::MakeMaker ;



my %WriteMakefileArgs = (
  "ABSTRACT" => "a library for sending email",
  "AUTHOR" => "Ricardo Signes <rjbs\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Email-Sender",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Email::Sender",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Email::Abstract" => "3.006",
    "Email::Address" => 0,
    "Email::Simple" => "1.998",
    "Fcntl" => 0,
    "File::Basename" => 0,
    "File::Path" => "2.06",
    "File::Spec" => 0,
    "IO::File" => 0,
    "IO::Handle" => 0,
    "List::MoreUtils" => 0,
    "Module::Runtime" => 0,
    "Moo" => "1.000008",
    "Moo::Role" => 0,
    "MooX::Types::MooseLike" => "0.15",
    "MooX::Types::MooseLike::Base" => 0,
    "Net::SMTP" => 0,
    "Scalar::Util" => 0,
    "Sub::Exporter" => 0,
    "Sub::Exporter::Util" => 0,
    "Sys::Hostname" => 0,
    "Throwable::Error" => "0.200003",
    "Try::Tiny" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "Capture::Tiny" => "0.08",
    "Cwd" => 0,
    "Exporter" => 0,
    "File::Temp" => 0,
    "Test::More" => "0.96",
    "lib" => 0
  },
  "VERSION" => "1.300016",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Capture::Tiny" => "0.08",
  "Carp" => 0,
  "Cwd" => 0,
  "Email::Abstract" => "3.006",
  "Email::Address" => 0,
  "Email::Simple" => "1.998",
  "Exporter" => 0,
  "Fcntl" => 0,
  "File::Basename" => 0,
  "File::Path" => "2.06",
  "File::Spec" => 0,
  "File::Temp" => 0,
  "IO::File" => 0,
  "IO::Handle" => 0,
  "List::MoreUtils" => 0,
  "Module::Runtime" => 0,
  "Moo" => "1.000008",
  "Moo::Role" => 0,
  "MooX::Types::MooseLike" => "0.15",
  "MooX::Types::MooseLike::Base" => 0,
  "Net::SMTP" => 0,
  "Scalar::Util" => 0,
  "Sub::Exporter" => 0,
  "Sub::Exporter::Util" => 0,
  "Sys::Hostname" => 0,
  "Test::More" => "0.96",
  "Throwable::Error" => "0.200003",
  "Try::Tiny" => 0,
  "lib" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



